### PR TITLE
Split truthy() into two functions

### LINKED ISSUE #1592: split truthy into separate functions
Right now, it also handles calls, specifically for dependency discovery:

https://github.com/rstudio/renv/blob/dd4f2b094e9ab1c79917f2e34b51f9975b5a0eef/R/dependencies.R#L769-L772

We should instead have a separate function for safely evaluating these expressions.

### PR DESCRIPTION
**Pull request by hadley**
* `truthy()` for env vars and options
* `renv_dependenciy_param_is_true()` for knitr chunk options

Fixes #1592

### CODE REVIEW

File: R/dependencies.R [hunk 1]
```diff
@@ -1707,6 +1707,22 @@ renv_dependencies_report <- function(errors) {
...
+
+renv_dependencies_param_is_true <- function(x, default = TRUE) {
+  # https://github.com/rstudio/renv/issues/1558
+  if (is.call(x) || is.symbol(x)) {
+    x <- tryCatch(renv_dependencies_eval(x), error = identity)
+    if (inherits(x, "error"))
+      return(default)
+  }
+
+  if (is.logical(x) && length(x) == 1 && !is.na(x)) {
```
Comment (kevinushey) [inline R/dependencies.R:1719]: For a chunk header like:

`````
```{r, renv.ignore=0}
# blah
```
`````

We would end up ignoring the value of `0` as it's not logical?

In other words, I think we should fall back to `truthy()` here instead for whatever value the user has supplied, or just coerce non-logical values to logical.

Comment (hadley) [reply R/dependencies.R:1719]: Sorry, I didn't realise knitr was so loosely typed here. Even `include = "false"` works 😭

Comment (kevinushey) [reply R/dependencies.R:1719]: "Be liberal in what you accept", as they say. 😉

Comment (hadley) [reply R/dependencies.R:1719]: FWIW `truthy()` isn't quite right either because `include = "True"` doesn't work. But we can coerce to logical.

**Pull request closed without merging**